tests/unit/test_charm.py

```diff
@@ -39,13 +40,13 @@ def test_init_postgresql_relation(self):
         # We'll only test the case where _stored already
         # has content. _stored being empty is basically tested
         # by all the other functions
-
-        with patch('test_charm.GunicornK8sCharm._stored') as mock_stored:
-            with patch('pgsql.PostgreSQLClient'):
-                with patch('test_charm.GunicornK8sCharm.on'):
-                    mock_stored.reldata = {'pg': 'foo'}
-                    c = GunicornK8sCharm(MagicMock())
-                    self.assertEqual(c._stored.reldata, mock_stored.reldata)
+        with patch("charm.GunicornK8sCharm.__init__") as mock_init, patch(
+            'test_charm.GunicornK8sCharm._stored'
+        ) as mock_stored, patch('pgsql.PostgreSQLClient'), patch('test_charm.GunicornK8sCharm.on'):
+            mock_stored.reldata = {'pg': 'foo'}
+            mock_init.return_value = None
+            c = GunicornK8sCharm(MagicMock())
```

Comment:
c, r... Try to avoid this short variable names (except for counters or similar cases). It makes the code a little difficult to read.